cli/src/main/java/com/devonfw/tools/ide/version/VersionRange.java

```diff
@@ -46,11 +158,17 @@ public VersionIdentifier getMax() {
   public boolean contains(VersionIdentifier version) {
 
     if (this.min != null) {
+      if (this.min.equals(version)) {
+        return !this.leftIsExclusive;
+      }
       if (version.isLess(this.min)) {
         return false;
       }
     }
     if (this.max != null) {
+      if (this.max.equals(version)) {
+        return !this.rightIsExclusive;
+      }
       if (version.isGreater(this.max)) {
         return false;
       }
```

Comment:
```suggestion
    if (this.min != null) {
      VersionComparisonResult compareMin = version.compareVersion(this.min);
      if (compareMin.isLess()) {
        return false;
      } else if (compareMin.isEqual() && !this.leftIsExclusive) {
        return false;
      }
    }
    if (this.max != null) {
      VersionComparisonResult compareMax = version.compareVersion(this.max);
      if (compareMax.isGreater()) {
        return false;
      } else if (compareMax.isEqual() && !this.rightIsExclusive) {
        return false;
      }
```

In case min equals, you did not check max anymore. This is a bug. It is always good to extend JUnits to cover these boundaries (https://en.wikipedia.org/wiki/Boundary-value_analysis).
Further if not equals the comparison was done twice that can be slightly optimized, by calling `compareVersion` only once for min/max.